### PR TITLE
me-17957: test if video on autoplay on scroll page is playing

### PR DESCRIPTION
**Pull request by ShayLevi**
Relevant task - https://cloudinary.atlassian.net/browse/ME-17957
This test is navigating to autoplay on scroll page (autoplay-on-scroll.html) and make sure that video is playing automatically on scroll.
First testing that the video is not playing (before scrolling), then scrolling until video element is visible and test that it is playing (checking that the video is not paused).

### CODE REVIEW

File: test/e2e/src/pom/autoplayOnScrollPage.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  /**
+   * Scrolls the page until the video element is visible using scrollIntoViewIfNeeded method.
+   * This action ensures the autoplay behavior is triggered when the video comes into view.
+   * It waits briefly after scrolling to give the video enough time to load and trigger autoplay.
+   */
+  async scrollToVideoElement() {
+    await this.page.locator(AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR).scrollIntoViewIfNeeded();
+    // Wait briefly to give time for the autoplay to apply
+    await this.page.waitForTimeout(1000);
```
Comment (refael-m) [inline test/e2e/src/pom/autoplayOnScrollPage.ts:24]: don't use waitForTimeout, use expect.toPass.

**Review by refael-m — CHANGES_REQUESTED** [review]
see comments

File: test/e2e/src/pom/autoplayOnScrollPage.ts (hunk 1 above)
Comment (ShayLevi) [reply test/e2e/src/pom/autoplayOnScrollPage.ts:24]: I removed the waitForTimeout and used expect.toPass in test step of validating that the video is auto playing after scrolling

File: test/e2e/src/pom/autoplayOnScrollPage.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    super(page);
+    this.autoplayOnScrollVideoComponent = new VideoComponent(page, AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR);
+  }
+
+  /**
+   * Scrolls the page until the video element is visible using scrollIntoViewIfNeeded method.
+   * This action ensures the autoplay behavior is triggered when the video comes into view.
+   */
+  async scrollToVideoElement() {
+      await this.page.locator(AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR).scrollIntoViewIfNeeded();
```
Comment (refael-m) [inline test/e2e/src/pom/autoplayOnScrollPage.ts:22]: you don't need it, you can use something like autoplayOnScrollVideoComponent. scrollIntoViewIfNeeded() in the test.

File: test/e2e/specs/autoplayOnScrollPage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  await test.step('Validating that the video is not playing before scrolling (in case isPause is true)', async () => {
+    expect(await pomPages.autoplayOnScrollPage.autoplayOnScrollVideoComponent.isPaused()).toEqual(true);
+  });
+  await test.step('Scroll until the video element is visible', async () => {
+    await pomPages.autoplayOnScrollPage.scrollToVideoElement();
+  });
+  await test.step('Validating that the video is auto playing after scrolling (in case isPause is false)', async () => {
+    await expect(async () => {
+      expect(await pomPages.autoplayOnScrollPage.autoplayOnScrollVideoComponent.isPaused()).toEqual(false);
+    }).toPass({ intervals: [2], timeout: 1000 });
```
Comment (refael-m) [inline test/e2e/specs/autoplayOnScrollPage.spec.ts:28]: why did you add 2 here?

**Review by refael-m — CHANGES_REQUESTED** [review]
see comments

File: test/e2e/src/pom/autoplayOnScrollPage.ts (hunk 2 above)
Comment (ShayLevi) [reply test/e2e/src/pom/autoplayOnScrollPage.ts]: Added BaseComponent and modified the videoComponent to extend it. Removed the scrollToVideoElement function and used scrollIntoViewIfNeeded() in the test.

File: test/e2e/specs/autoplayOnScrollPage.spec.ts (hunk 3 above)
Comment (ShayLevi) [reply test/e2e/specs/autoplayOnScrollPage.spec.ts]: Changed it to intervals: [500], timeout: 3000

**Review by refael-m — APPROVED** [review]
